src/model/remote-builder/remote-builder.ts

```diff
@@ -0,0 +1,422 @@
+import { customAlphabet } from 'nanoid';
+import AWSBuildPlatform from './aws-build-platform';
+import * as core from '@actions/core';
+import RemoteBuilderConstants from './remote-builder-constants';
+import { BuildParameters } from '..';
+const repositoryDirectoryName = 'repo';
+const efsDirectoryName = 'data';
+const cacheDirectoryName = 'cache';
+
+class RemoteBuilder {
+  static SteamDeploy: boolean = false;
+  static async build(buildParameters: BuildParameters, baseImage) {
+    try {
+      this.SteamDeploy = process.env.STEAM_DEPLOY !== '' || false;
+      const nanoid = customAlphabet(RemoteBuilderConstants.alphabet, 4);
+      const buildUid = `${process.env.GITHUB_RUN_NUMBER}-${buildParameters.platform
+        .replace('Standalone', '')
+        .replace('standalone', '')}-${nanoid()}`;
+      const defaultBranchName =
+        process.env.GITHUB_REF?.split('/')
+          .filter((x) => {
+            x = x[0].toUpperCase() + x.slice(1);
+            return x;
+          })
+          .join('') || '';
+      const branchName = process.env.REMOTE_BUILDER_CACHE !== '' ? process.env.REMOTE_BUILDER_CACHE : defaultBranchName;
+      const token: string = buildParameters.githubToken;
+      const defaultSecretsArray = [
+        {
+          ParameterKey: 'GithubToken',
+          EnvironmentVariable: 'GITHUB_TOKEN',
+          ParameterValue: token,
+        },
+      ];
+      // await RemoteBuilder.ProvisionPlatformResources();
+      await RemoteBuilder.SetupStep(buildUid, buildParameters, branchName, defaultSecretsArray);
+      await RemoteBuilder.BuildStep(buildUid, buildParameters, baseImage, defaultSecretsArray);
+      await RemoteBuilder.CompressionStep(buildUid, buildParameters, branchName, defaultSecretsArray);
+      await RemoteBuilder.UploadArtifacts(buildUid, buildParameters, branchName, defaultSecretsArray);
+      if (this.SteamDeploy) {
+        await RemoteBuilder.DeployToSteam(buildUid, buildParameters, defaultSecretsArray);
+      }
+    } catch (error) {
+      core.setFailed(error);
+      core.error(error);
+    }
+  }
+  // static async ProvisionPlatformResources() {
+  //   await AWSBuildPlatform.setupPlatformResources();
+  // }
```

Comment:
Probably best to remove or add a comment above it what the plan for this snippet is.